psutil/_psaix.py

```diff
@@ -511,9 +511,11 @@ def status(self):
         # XXX is '?' legit? (we're not supposed to return it anyway)
         return PROC_STATUSES.get(code, '?')
 
-    def open_files(self):
+    def open_files(self, only_regular=True):
         # TODO rewrite without using procfiles (stat /proc/pid/fd/* and then
         # find matching name of the inode)
+        # <<< TODO >>>: does the current implementation ensure we only return
+        #               regular files? or is that missing?
```

Comment:
I could use some help here. Is it possible the existing implementation returns non-regular files already? Reading through the docs of [`procfiles`](https://www.ibm.com/docs/sk/aix/7.2?topic=p-procfiles-command), it mentions non-regular files:

> Regular files have permission based on mode it was opened with. Any non-regular files have 0 access mode.

So possibly we just weren't doing the filtering here and the fix is to add the missing filtering if `only_regular == True`?